__tests__/views/pages/harvesting/HarvestedBlocksList.spec.ts

```diff
@@ -0,0 +1,177 @@
+    const mockMosaicStore = {
+        namespaced: true,
+        state: {
+            networkCurrency: networkCurrency,
+        },
+        getters: {
+            networkCurrency: (state) => state.networkCurrency,
+        },
+    };
+
+    const renderComponent = (stateChanges?: { [field: string]: any }, dispatch?: () => any) => {
+        const localVue = createLocalVue();
+        localVue.use(Vuex);
+        localVue.use(iView);
+        const store = getStore(
+            {
+                account: mockAccountStore,
+                harvesting: mockHarvestingStore,
+                mosaic: mockMosaicStore,
+            },
+            stateChanges,
+            dispatch,
+            true,
+            undefined,
+            false,
+        );
+
+        return {
+            ...render(HarvestedBlocksList, {
+                store,
+                i18n,
+                localVue,
+            }),
+            store,
+        };
+    };
+
+    test('renders table with no data', async () => {
+        // Arrange + Act:
+        const { getByText } = renderComponent();
+
+        // Assert:
+        expect(getByText(i18n.t('block').toString())).toBeDefined();
+        expect(getByText(i18n.t('fees_earned').toString())).toBeDefined();
+        expect(getByText(i18n.t('no_harvested_blocks_yet').toString())).toBeDefined();
+    });
+
+    test('renders table with block rewards data', async () => {
+        // Arrange + Act:
+        const harvestedBlocks = [{ blockNo: UInt64.fromUint(2_000), fee: UInt64.fromUint(1_000_000) }];
+        const { getByText } = renderComponent({ harvestedBlocks });
+
+        // Assert:
+        expect(getByText('2000')).toBeDefined();
+        expect(getByText('1.0')).toBeDefined();
+    });
+
+    test('shows next page in the table when next page button is clicked', async () => {
+        // Arrange:
+        const dispatch = jest.fn();
+        const harvestedBlocks = [...Array(40).keys()].map((i) => ({
+            blockNo: UInt64.fromUint((i + 1) * 1_000),
+            fee: UInt64.fromUint((i + 1) * 1_000_000),
+        }));
+        const { getByTestId, getByText, queryByText, store } = renderComponent({ harvestedBlocks: harvestedBlocks.slice(0, 20) }, dispatch);
+
+        await Vue.nextTick();
+        // assert initial state
+        expect(getByText('1000')).toBeDefined();
+        expect(queryByText('21000')).toBeNull();
+
+        dispatch.mockImplementationOnce((type: string) => {
+            if (type === 'harvesting/LOAD_HARVESTED_BLOCKS') {
+                store.commit('harvesting/harvestedBlocks', harvestedBlocks.slice(20, 40));
+                store.commit('harvesting/harvestedBlocksPageInfo', { pageNumber: 2, isLastPage: true });
+            }
+        });
+
+        // Act:
+        getByTestId('btn-next-page').click();
+        await Vue.nextTick();
+
+        // // Assert:
+        expect(queryByText('1000')).toBeNull();
+        expect(getByText('21000')).toBeDefined();
```

Comment:
is it possible to search element by class or id?

Reply:
using css selectors or IDs are kind of anti-pattern in testing library philosophy (event test-id's are considered as escape hatch), in this case since we use iview library's Table component, we couldn't add any test-id's so went with the text query
> One of the guiding principles of the Testing Library APIs is that they should enable you to test your app the way your users use it